### DISCUSSION TITLE
A "chat" application using AT

### DISCUSSION THREAD

**Discussion by gleicon** [bluesky-social/atproto#3069] [How to do it?]
Hi, I'm looking to build my go-to learning app on feed protocols, a simple single channel chat app. I've already installed my own pds and tinkered with AT libraries but I'm having a hard time wrapping my head around on how to build a simple, private chat around it. Did anyone explored it ? I feel that the concept of DID, local repository and feed are already what I need but there is a component I'm not sure I have the rightview: what would be the "channel" if it was an IRC server. TIA for any tips, help and discouragements if that's too far fetched :) !

**Comment by yamarten** [comment]
Currently, sharing private content in atproto is out of scope. Future support is planned, but it's probably pretty far away. The current alternative is to use the [chat API](https://docs.bsky.app/docs/api/chat-bsky-convo-send-message) (DM) or make it yourself.

Please note that it is not PDS that usually has those APIs. In the case of the chat API, PDS only performs authentication and proxy, and does not store or recognize messages.

**Reply by gleicon** [reply]
Thanks, I didn't knew about chat api. I figured that if I had a non-federated PDS server, it would be a public (in the sense of not being p2p) but private network (in the sense of no one outside of my private network seeing it), in the same vein as the age old Yammer used to do. I also thought that the basic of the AT proto of having a feed and your personal data locally would be ideal to build a sort of decentralized team chat.
